File: c++/fibonacci.cpp
#include <iostream>
#include <vector>
using namespace std;

int get_factors_count(long long int number) {
    int count = 0;
    for (int i = 1; i * i <= number; ++i) {
        if (number % i == 0) {
            count += 2;
            if (i * i == number) {
                --count;
            }
        }
    }
    return count;
}

int main() {
    long long int a = 1, b = 1, temp = 0;
    while (true) {
        temp = a; 
        a = b; 
        b = temp + b;
        if (get_factors_count(a) >= 1000) {
            cout << a << endl;
            return 0;
        }
    }
}

File: c-sharp/fibonacci.cs
using System;
using System.Collections;

class Fibonacci() {
	public static void Main() {
		Int64 a = 1, b = 1;
		while (true)
		{
			Int64 temp = a; a = b; b = temp + b;
			if(factores(a) >= 1000) {
				Console.WriteLine(a);
				return ;
			}
		}
	}

	public static int factores(Int64 numero) {
		ArrayList divisores = new ArrayList();
		int max = (int)Math.Sqrt(numero);
		for(int factor = 1; factor <= max; ++factor) {
			if(numero % factor == 0) {
				divisores.Add(factor);
				if(factor != numero/factor) {
					divisores.Add(numero/factor);
				}
			}
		}
		return divisores.Count;
	}
}